src/glang/codegen/builtin_callables.py

```diff
@@ -225,25 +240,28 @@ class BitwiseXorExpression(ArithmeticExpression):
     USER_FACING_NAME = "__builtin_bitwise_xor"
 
 
-class CompareExpression(BasicIntegerExpression):
+class CompareExpression(BasicNumericExpression):
     @staticmethod
     def get_result_type(arguments: list[TypedExpression]) -> Type:
         return BoolType()
 
 
 class IsEqualExpression(CompareExpression):
+    FLOATING_POINT_IR = "fcmp oeq"  # Ordered and equal (can we relax this?)
```

Comment:
Do the ordered and unordered versions produce different assembly? Unordered is presumably faster, although ordered might produce slightly saner code?

Not even sure how a QNAN would come about though

Reply:
I've looked at some x86 assembly and you're right, the unordered versions produce faster code. I've spent some time thinking, and I think it is fine to stop requiring the order: we're already ignoring integer overflows so why not also ignore NaNs.